### PR TITLE
POL-239 - Google Compute Engine (GCE) Plugin all changes

### PR DESCRIPTION
**Pull request by gopinathmahakud1**
Google Compute Engine (GCE) Plugin all changes

### Description

Updated the Google Compute Engine Plugin to work with the Resource Browser/Plugin Registration

### Issues Resolved

Updated required changes.

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

image

image

### CODE REVIEW

File: google/gce/gce_plugin.rb [hunk 1]
```diff
@@ -86,7 +117,20 @@
       path "/projects/$project/regions/$region/addresses"
       type "address"
       output_path "items"
+          field "page_size" do
+          type 'string'
+        location 'query'
+        alias_for 'maxResults'
+      end
+     pagination $aws_pagination
```
Comment (cdwilhelm) [inline google/gce/gce_plugin.rb]: should be google_pagination

File: google/gce/gce_plugin.rb [hunk 2]
```diff
@@ -544,6 +601,19 @@
       path "/projects/$project/zones/$zone/disks"
       type "disk"
       output_path "items"
+      field "page_size" do
+          type 'string'
+          location 'query'
+          alias_for 'maxResults'
+      end
+      pagination $aws_pagination
```
Comment (cdwilhelm) [inline google/gce/gce_plugin.rb]: google_pagination

File: google/gce/gce_plugin.rb [hunk 3]
```diff
@@ -884,6 +954,19 @@
       path "/projects/$project/global/forwardingRules"
       type "forwardingRule"
       output_path "items"
+      field "page_size" do
+          type 'string'
+          location 'query'
+          alias_for 'maxResults'
+      end
+      pagination $aws_pagination
```
Comment (cdwilhelm) [inline google/gce/gce_plugin.rb]: google_pagination

File: google/gce/gce_plugin.rb [hunk 4]
```diff
@@ -1560,6 +1656,19 @@
       path "/projects/$project/zones/$zone/instanceGroups"
       type "instanceGroup"
       output_path "items"
+        field "page_size" do
+          type 'string'
+          location 'query'
+          alias_for 'maxResults'
+      end
+      pagination $aws_pagination
```
Comment (cdwilhelm) [inline google/gce/gce_plugin.rb]: google_pagination

File: google/gce/gce_plugin.rb [hunk 5]
```diff
@@ -1794,7 +1903,20 @@
       path "/projects/$project/zones/$zone/instances"
       type "instance"
       output_path "items"
+      field "page_size" do
+          type 'string'
+          location 'query'
+          alias_for 'maxResults'
+      end
+      pagination $aws_pagination
```
Comment (cdwilhelm) [inline google/gce/gce_plugin.rb]: google_pagination

File: google/gce/gce_plugin.rb (hunk 1 above)
Comment (gopinathmahakud1) [reply google/gce/gce_plugin.rb]: I have addressed

File: google/gce/gce_plugin.rb (hunk 2 above)
Comment (gopinathmahakud1) [reply google/gce/gce_plugin.rb]: I have addressed

File: google/gce/gce_plugin.rb (hunk 3 above)
Comment (gopinathmahakud1) [reply google/gce/gce_plugin.rb]: I have addressed

File: google/gce/gce_plugin.rb (hunk 4 above)
Comment (gopinathmahakud1) [reply google/gce/gce_plugin.rb]: I have addressed

File: google/gce/gce_plugin.rb (hunk 5 above)
Comment (gopinathmahakud1) [reply google/gce/gce_plugin.rb]: I have addressed

**Comment by cdwilhelm** [comment]
@gopinathmahakud1   changes look good.  thank you

**Comment by cdwilhelm** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">google/gce/changelog.md:3: MD002 First header should be a top level header<br />
google/gce/changelog.md:8: MD022 Headers should be surrounded by blank lines<br />
google/gce/changelog.md:10: MD032 Lists should be surrounded by blank lines</p>

<p>A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Should this include readme changes</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: google/gce/gce_plugin.rb [hunk 6]
```diff
@@ -28,7 +63,7 @@
 
   # This resource was generated using the documentation from https://cloud.google.com/compute/docs/reference/latest/addresses.
   type "address" do
-    href_templates "{{selfLink}}","{{items[*].selfLink}}","{{items.*.addresses[].selfLink}}"
+    href_templates "{{.id}}","{{.items| .[].id}}","{{.items | .[].addresses // [] | .[].id}}"
```
Comment (rshade) [inline google/gce/gce_plugin.rb:66]: @gopinathmahakud1 please revert these to href_templates, that are compatible with all actions. @cdwilhelm  this doesn't need to break existing functionality.

**Comment by rshade** [comment]
Closing this as OpenAPI Plugins are now preferred.

**Pull request closed without merging**